Build with .NET 10 successfully

jonathanpeppers opened the pull request:
Changes required:

* Get appropriate `net9` branded workload manifests from dotnet/runtime

* Fix NuGet package downgrades:

```
    src\SingleProject\Resizetizer\src\Resizetizer.csproj : error NU1605:
      Warning As Error: Detected package downgrade: System.Runtime.CompilerServices.Unsafe from 6.1.0 to 6.0.0. Reference the package directly from the project to select a different version.
       Microsoft.Maui.Resizetizer -> Microsoft.Bcl.AsyncInterfaces 10.0.0-alpha.1.25058.4 -> System.Threading.Tasks.Extensions 4.6.0 -> System.Runtime.CompilerServices.Unsafe (>= 6.1.0)
       Microsoft.Maui.Resizetizer -> System.Runtime.CompilerServices.Unsafe (>= 6.0.0)
    src\Core\src\Core.csproj : error NU1605:
      Warning As Error: Detected package downgrade: System.Numerics.Vectors from 4.6.0 to 4.5.0. Reference the package directly from the project to select a different version.
       Microsoft.Maui.Core -> Microsoft.Extensions.Logging.Abstractions 10.0.0-alpha.1.25057.17 -> System.Memory 4.6.0 -> System.Numerics.Vectors (>= 4.6.0)
       Microsoft.Maui.Core -> System.Numerics.Vectors (>= 4.5.0)
```

* Update to iOS, tvOS, MacCatalyst 18.2

```
    10.0.100-alpha.1.25059.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(259,5):
    error NETSDK1140:
        18.0 is not a valid TargetPlatformVersion for MacCatalyst. Valid versions include:
        18.2
    10.0.100-alpha.1.25059.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(259,5):
    error NETSDK1140:
        18.0 is not a valid TargetPlatformVersion for iOS. Valid versions include:
        18.2
```

* Temporarily ignore `CS9270`:
  * https://github.com/dotnet/roslyn/pull/76312
  * https://github.com/dotnet/roslyn/issues/72133

```
    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-BindableLayout-BindableLayout.cs-GeneratedBindingInterceptors-235-10.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)
    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-ContentConverter.cs-GeneratedBindingInterceptors-72-13.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)
    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-ContentConverter.cs-GeneratedBindingInterceptors-77-13.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)
```

* iOS API changes:

```
    error CS0618: 'CLLocation.AccurracyBestForNavigation' is obsolete: 'Use 'AccuracyBestForNavigation' instead.'
```
```
    Screenshot.ios.cs(57,69): error CS8602: Dereference of a possibly null reference.
```
```
    Platform\iOS\KeyboardAutoManagerScroll.cs(867,55): error CS8602: Dereference of a possibly null reference.
```
```
    src\Core\src\Platform\iOS\MauiSwipeView.cs(601,8): error CS8625: Cannot convert null literal to non-nullable reference type.
    src\Core\src\Platform\iOS\MauiSwipeView.cs(634,8): error CS8625: Cannot convert null literal to non-nullable reference type.
```
```
    ResignFirstResponderTouchGestureRecognizer.iOS.cs(70,12): error CS8600: Converting null literal or possible null value to non-nullable type.
```
```
    GesturePlatformManager.iOS.cs(282,21): error CS8602: Dereference of a possibly null reference.
```

Most notably:

* `UIView.Window` can be null.

* Typo in API fixed `AccurracyBestForNavigation` -> `AccuracyBestForNavigation`

* Ignore warnings related with trimming 
  `IL2026; IL2067;IL2072 ;IL2075;IL2087; IL2091;`
  
* Move to Xcode 16.2

jonathanpeppers committed:
Build with .NET 10 successfully

Changes required:

* Get appropriate `net9` branded workload manifests from dotnet/runtime

* Fix NuGet package downgrades:

    src\SingleProject\Resizetizer\src\Resizetizer.csproj : error NU1605:
      Warning As Error: Detected package downgrade: System.Runtime.CompilerServices.Unsafe from 6.1.0 to 6.0.0. Reference the package directly from the project to select a different version.
       Microsoft.Maui.Resizetizer -> Microsoft.Bcl.AsyncInterfaces 10.0.0-alpha.1.25058.4 -> System.Threading.Tasks.Extensions 4.6.0 -> System.Runtime.CompilerServices.Unsafe (>= 6.1.0)
       Microsoft.Maui.Resizetizer -> System.Runtime.CompilerServices.Unsafe (>= 6.0.0)
    src\Core\src\Core.csproj : error NU1605:
      Warning As Error: Detected package downgrade: System.Numerics.Vectors from 4.6.0 to 4.5.0. Reference the package directly from the project to select a different version.
       Microsoft.Maui.Core -> Microsoft.Extensions.Logging.Abstractions 10.0.0-alpha.1.25057.17 -> System.Memory 4.6.0 -> System.Numerics.Vectors (>= 4.6.0)
       Microsoft.Maui.Core -> System.Numerics.Vectors (>= 4.5.0)

* Update to iOS, tvOS, MacCatalyst 18.2

    10.0.100-alpha.1.25059.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(259,5):
    error NETSDK1140:
        18.0 is not a valid TargetPlatformVersion for MacCatalyst. Valid versions include:
        18.2
    10.0.100-alpha.1.25059.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(259,5):
    error NETSDK1140:
        18.0 is not a valid TargetPlatformVersion for iOS. Valid versions include:
        18.2

* Temporarily ignore `CS9270`:
  * https://github.com/dotnet/roslyn/pull/76312
  * https://github.com/dotnet/roslyn/issues/72133

    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-BindableLayout-BindableLayout.cs-GeneratedBindingInterceptors-235-10.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)
    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-ContentConverter.cs-GeneratedBindingInterceptors-72-13.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)
    artifacts\obj\Controls.Core\Debug\net10.0\Microsoft.Maui.Controls.BindingSourceGen\Microsoft.Maui.Controls.BindingSourceGen.BindingSourceGenerator\D--src-maui-src-Controls-src-Core-ContentConverter.cs-GeneratedBindingInterceptors-77-13.g.cs(43,4): error CS9270: 'InterceptsLocationAttribute(string, int, int)' is not supported. Move to 'InterceptableLocation'-based generation of these attributes instead. (https://github.com/dotnet/roslyn/issues/72133)

* iOS API changes:

    error CS0618: 'CLLocation.AccurracyBestForNavigation' is obsolete: 'Use 'AccuracyBestForNavigation' instead.'

    Screenshot.ios.cs(57,69): error CS8602: Dereference of a possibly null reference.

    Platform\iOS\KeyboardAutoManagerScroll.cs(867,55): error CS8602: Dereference of a possibly null reference.

    src\Core\src\Platform\iOS\MauiSwipeView.cs(601,8): error CS8625: Cannot convert null literal to non-nullable reference type.
    src\Core\src\Platform\iOS\MauiSwipeView.cs(634,8): error CS8625: Cannot convert null literal to non-nullable reference type.

    ResignFirstResponderTouchGestureRecognizer.iOS.cs(70,12): error CS8600: Converting null literal or possible null value to non-nullable type.

    GesturePlatformManager.iOS.cs(282,21): error CS8602: Dereference of a possibly null reference.

Most notably:

* `UIView.Window` can be null.

* Typo in API fixed `AccurracyBestForNavigation` -> `AccuracyBestForNavigation`

jonathanpeppers committed:
Merge branch 'net10.0' into dev/peppers/net10.0

jonathanpeppers commented:
/azp run

rmarinho commented:
/azp run

jonathanpeppers committed:
Disable tizen net10.0 temporarily

jonathanpeppers committed:
TargetFramework=net10.0

jonathanpeppers committed:
TargetFramework=net10.0

jonathanpeppers committed:
Unused yaml: `TestTargetFrameworks`

jonathanpeppers committed:
Update RunnerGenerator.cs

Workaround: https://github.com/dotnet/android/commit/715a36af86c68bac623a4174477165bb14b4c371

jonathanpeppers committed:
Put iOS simulator versions back to 18.0

jonathanpeppers committed:
Update RunnerGenerator.cs

jonathanpeppers committed:
Update xUnitSharedAttributes.cs

    D:\src\maui\src\TestUtils\src\TestShared\xUnitSharedAttributes.cs(75,89): error CS0117: 'DynamicallyAccessedMemberTypes' does not contain a definition for 'PublicParameterlessConstructors'
    D:\src\maui\src\TestUtils\src\TestShared\xUnitSharedAttributes.cs(83,63): error CS0117: 'DynamicallyAccessedMemberTypes' does not contain a definition for 'PublicParameterlessConstructors'

jonathanpeppers committed:
Update Core.DeviceTests.Shared.csproj

jonathanpeppers committed:
Update RunnerGenerator.cs

jonathanpeppers committed:
Remove Tizen project

jonathanpeppers committed:
Update ios.cake

jonathanpeppers commented on src/TestUtils/src/DeviceTests.Runners.SourceGen/RunnerGenerator.cs:
```diff
@@ -97,12 +97,15 @@ string GenerateAndroidSource()
 			return @"
 #if !SKIP_RUNNER_ENTRYPOINT_GENERATION && !SKIP_VISUAL_RUNNER_ENTRYPOINT_GENERATION && !SKIP_VISUAL_RUNNER_APPLICATION_GENERATION
+using System;
+using JniHandleOwnership = Android.Runtime.JniHandleOwnership;
+
 namespace " + RootNamespace + @"
 {
 	[global::Android.App.Application]
 	partial class " + appName + @" : global::Microsoft.Maui.MauiApplication
 	{
-		public " + appName + @"(global::System.IntPtr handle, global::Android.Runtime.JniHandleOwnership ownership)
+		public " + appName + @"(IntPtr handle, JniHandleOwnership ownership)
```
The changes here, we are going to fix in:
* https://github.com/dotnet/android/pull/9680

`global::` didn't work in this new analyzer.

jonathanpeppers committed:
Update Core.DeviceTests.csproj

rmarinho committed:
Update android

rmarinho committed:
Update xcode

rmarinho committed:
Disable trimming warnings for now

rmarinho committed:
[iOS] Move to 18.2 simulators

rmarinho committed:
Ignore more warnings

rmarinho committed:
Ignore more warnings

rmarinho committed:
Update MainViewController.cs

rmarinho committed:
Ignore more warnings

jonathanpeppers commented on src/TestUtils/src/TestShared/xUnitSharedAttributes.cs:
```diff
@@ -71,14 +72,15 @@ public class ClassDataAttribute : DataAttribute
 		/// </summary>
 		/// <param name="class">The class that provides the data.</param>
-		public ClassDataAttribute(Type @class)
+		public ClassDataAttribute([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type @class)
```
Let me check `TestUtils.csproj::TargetFramework=netstandard2.0]`

Maybe I can revert this change, I found there were so many trimmer warnings in some test projects I just disabled the analyzer.

rmarinho replied:
oh it doesn't t exist on netstandard2.0

jonathanpeppers replied:
I think instead I'm going to globally do this:
```xml
<!-- FIXME: various test projects have trimmer warnings -->
<EnableTrimAnalyzer Condition=" '$(IsTestProject)' == 'true' ">false</EnableTrimAnalyzer>
```
Shouldn't risk us shipping trimmer warnings this way, and just ignore them in test projects.

rmarinho committed:
Disable more warnings

jonathanpeppers commented on Directory.Build.targets:
```diff
@@ -36,7 +36,7 @@
       Temporarily disable trimming warnings
     -->
-    <NoWarn>$(NoWarn);IL2026;IL2075;IL2067;IL2091;IL2087;IL2072</NoWarn>
+    <NoWarn>$(NoWarn);IL2026;IL2057;IL2062;IL2067;IL2072;IL2075;IL2087;IL2091;</NoWarn>
```
I wouldn't recommend doing it this way.

rmarinho committed:
Update xUnitSharedAttributes.cs

jonathanpeppers committed:
Put trimmer warnings back

jonathanpeppers committed:
EnableTrimAnalyzer=false for test projects across repo

jonathanpeppers committed:
Make `EnableTizen()` a no-op

jonathanpeppers committed:
Update Controls.DeviceTests.csproj

jonathanpeppers committed:
Install `mobile-librarybuilder-net9`

Context: https://github.com/xamarin/xamarin-macios/pull/21848/commits/cb1340221774ba9af226ac71a5aef551d765ee21

jonathanpeppers committed:
Update WindowsTemplateTest.cs

jonathanpeppers committed:
`mobile-librarybuilder-net9` on test lanes

jonathanpeppers committed:
Update Essentials.Sample.csproj

jonathanpeppers committed:
See if newer .NET SDK fixes maui-blazor

Context: https://github.com/dotnet/sdk/compare/26fd6d135ae2605cf618caeffe83b3be5e314ac5...60e9a46917ff91cc570272ecc83249497c99a79f

I see a dotnet/aspnetcore bump in the diff.

jonathanpeppers committed:
Essentials.Sample.csproj is a "sample project"

jonathanpeppers committed:
> darc update-dependencies --id 252455

jonathanpeppers committed:
Revert "Update RunnerGenerator.cs"

This reverts commit b9f32f6b0037a48e9833c72a9ab2c1840e18e788.

jonathanpeppers committed:
Revert "Update RunnerGenerator.cs"

This reverts commit 9f6d36dff6f4c878f3c97ed47dc46655ede8e5e4.

jonathanpeppers committed:
Revert "Update RunnerGenerator.cs"

This reverts commit 0f43f7def0e4df427e589814de674c8a314d1d50.

jonathanpeppers committed:
darc update-dependencies --id 252613

jonathanpeppers committed:
darc update-dependencies --coherency-only

jonathanpeppers committed:
Update Version.Details.xml

jonathanpeppers committed:
SuppressTrimAnalysisWarnings=true for test/sample projects

rmarinho committed:
Update Controls.DeviceTests.csproj

rmarinho committed:
Revert "Update Controls.DeviceTests.csproj"

This reverts commit 5ce03907e3701932a8678c2d1d0687b871a66b28.

rmarinho committed:
Disable iOS test for now

rmarinho committed:
Update provision.yml

rmarinho committed:
Update versions

rmarinho committed:
Update versions

rmarinho committed:
Revert "Update provision.yml"

This reverts commit 33a8ee569ca6d31950829375f4be05aad1235c00.

rmarinho committed:
Update LabelExtensions.cs

rmarinho committed:
Update ButtonTests.iOS.cs

rmarinho committed:
Try these new aspnet versions

rmarinho committed:
Update AssemblyInfoTests.cs

jonathanpeppers committed:
$(TrimmerSingleWarn)=false

jonathanpeppers committed:
Update MauiBlazorWebView.DeviceTests.csproj

jonathanpeppers committed:
$(IsTestProject) -> $(MauiTestProject)

The old name conflicts with dotnet/arcade.

jonathanpeppers committed:
Update Core.UnitTests.csproj

jonathanpeppers committed:
Update MacTemplateTest.cs

jonathanpeppers committed:
Update MacTemplateTest.cs

jonathanpeppers committed:
Revert "Update MacTemplateTest.cs"

This reverts commit 7ed2b89e645bfa996073c0d3990fbd7d276ceb13.

jonathanpeppers committed:
Revert "Update MacTemplateTest.cs"

This reverts commit 882c45e1eeed5d8afc712ddae1d99d6b76632f4f.

jonathanpeppers committed:
EnableTrimAnalyzer=false for default projects

Context: https://github.com/xamarin/xamarin-macios/pull/21351

This will avoid some of the fallout of the above PR.

jonathanpeppers committed:
$(EnableTrimAnalyzer) take 2

jonathanpeppers committed:
Update TestUtils.DeviceTests.Sample.csproj

jonathanpeppers committed:
Update Microsoft.Maui.Controls.Common.targets

jonathanpeppers committed:
Update Microsoft.Maui.Controls.Common.targets

jonathanpeppers committed:
Update SimpleTemplateTest.cs

jonathanpeppers committed:
Update Microsoft.Maui.Controls.Common.targets

jonathanpeppers committed:
System.Text.Json.Serialization.Converters.EnumConverterFactory warning is gone

jonathanpeppers committed:
Potential fix for M.E.DI failure

Fixes?

    [15:12:46.2484560] 2025-01-22 15:12:46.242709-0800 RunOniOSmauiRel1586373188[97228:54445246] Received unhandled Objective-C exception that was marshalled from a managed exception: A suitable constructor for type 'Microsoft.Maui.Hosting.FontsMauiAppBuilderExtensions+FontInitializer' could not be located. Ensure the type is concrete and services are registered for all parameters of a public constructor. (System.InvalidOperationException)
[15:12:46.2484780]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache, ServiceIdentifier, Type, CallSiteChain) + 0x4de
[15:12:46.2485210]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor, ServiceIdentifier, CallSiteChain, Int32) + 0x1a5
[15:12:46.2485640]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateEnumerable(ServiceIdentifier, CallSiteChain) + 0x42f
[15:12:46.2485930]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(ServiceIdentifier serviceIdentifier, CallSiteChain callSiteChain) + 0xfb
[15:12:46.2486100]    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(ServiceIdentifier, CallSiteChain) + 0x92
[15:12:46.2486220]    at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(ServiceIdentifier serviceIdentifier) + 0x68
[15:12:46.2486330]    at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey, Func`2) + 0xdc
[15:12:46.2486790]    at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier, ServiceProviderEngineScope) + 0x39
[15:12:46.2486900]    at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider, Type) + 0x3d
[15:12:46.2487000]    at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider) + 0x2f
[15:12:46.2487100]    at Microsoft.Maui.MauiContextExtensions.InitializeAppServices(MauiApp) + 0x1b
[15:12:46.2487190]    at Microsoft.Maui.Hosting.MauiAppBuilder.Build() + 0x67
[15:12:46.2487420]    at Microsoft.Maui.MauiUIApplicationDelegate.WillFinishLaunching(UIApplication application, NSDictionary launchOptions) + 0x44
[15:12:46.2487530]    at RunOniOSmauiRel1586373188!<BaseAddress>+0xe200a

jonathanpeppers committed:
Merge branch 'net10.0' into dev/peppers/net10.0

rmarinho committed:
Merge branch 'net10.0' into dev/peppers/net10.0

jonathanpeppers committed:
Revert "Potential fix for M.E.DI failure"

This reverts commit 802e4c15674612e0af2bc4e07fefd63a8b622554.

rmarinho committed:
Update versions with CoherentParentDependency

rmarinho committed:
Update versions with coherent Microsoft.JSInterop

jonathanpeppers committed:
Use strongly-typed Transient() for M.E.DI

jonathanpeppers committed:
MauiVersionCurrent / MauiVersionPrevious

jonathanpeppers committed:
Update FontsMauiAppBuilderExtensions.cs

jonathanpeppers committed:
Revert "Update FontsMauiAppBuilderExtensions.cs"

This reverts commit 6b8e48ad907cb184fd6e5b471bbe290a309cfdb8.

jonathanpeppers committed:
Revert "Use strongly-typed Transient() for M.E.DI"

This reverts commit b0e0e45fd76d5645e480321f42f56386a1285384.

jonathanpeppers committed:
Comment a few NativeAOT tests
